README.md

```diff
@@ -0,0 +1,54 @@
+# Data Source Guide Example - Weather API
+
+A simple weather API built with FastAPI. This API provides random weather data based on latitude and longitude coordinates.
+
+## Requirements
+
+- [Pre-commit](https://pre-commit.com/#install)
+- [Python 3.11+](https://www.python.org/downloads/)
+- [Poetry](https://python-poetry.org/docs/#installation)
+
+To set up the `pre-commit` hooks, run `pre-commit install` in the repo. After it you can
+manually run `pre-commit` only for your changes or `pre-commit run --all-files` for all
+files.
+
+### Install Dependencies
+
+To install dependencies, first set up your environment with Poetry:
+
+````bash
```

Comment:
Please check the formatting; you have 4 backticks instead of 3 and I believe the ending one is missing.